customer-terminate-service/domain/src/test/java/se/inera/intyg/cts/domain/repository/InMemoryTerminationRepository.java

```diff
@@ -0,0 +1,37 @@
+package se.inera.intyg.cts.domain.repository;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import se.inera.intyg.cts.domain.model.Termination;
+import se.inera.intyg.cts.domain.model.TerminationId;
+import se.inera.intyg.cts.domain.model.TerminationStatus;
+
+public class InMemoryTerminationRepository implements TerminationRepository {
```

Comment:
Reason why I didn't use mocking is that I don't want the tests to be too coupled to implementation. Instead by providing simple inMemory implementation of the repositories to remove any dependencies to databases or external systems the unit tests can easily add needed testdata and fokus on verifying the behaviour.

I think Mockito is great and have used it extensively, but it is not uncommon that wiring up the tests have to deal many implementation details which easily break when you refactor. I'm trying this approach instead, when writing the code from scratch, and aiming for high testability. If it shows that the approach doesn't work, which I don't see right now, I will rewrite the tests with mocks.

Reply:
As with InMemoryCertificateRepository  i do not see the benefit of adding untested implementations this class also adds its own logic for findByTerminationId and findByStatuses. This adds complexity and makes it unsafe due to the fact that it is used in both CollectExportContentTest and ExportPackageTest. Instead of reading the test we must check the implementation to see what will be returned and a change in the implementations affects multiple test classes.